src/Shared/StackTrace/ExceptionDetails/ExceptionDetailsProvider.cs

```diff
@@ -30,15 +33,27 @@ public IEnumerable<ExceptionDetails> GetDetails(Exception exception)
                 yield return new ExceptionDetails
                 {
                     Error = ex,
-                    StackFrames = StackTraceHelper.GetFrames(ex)
-                            .Select(frame => GetStackFrameSourceCodeInfo(
-                                frame.MethodDisplayInfo.ToString(),
-                                frame.FilePath,
-                                frame.LineNumber))
+                    StackFrames = GetStackFrames(ex),
                 };
             }
         }
 
+        private IEnumerable<StackFrameSourceCodeInfo> GetStackFrames(Exception original)
+        {
+            var stackFrames = StackTraceHelper.GetFrames(original, out var exception)
+                .Select(frame => GetStackFrameSourceCodeInfo(
+                    frame.MethodDisplayInfo.ToString(),
+                    frame.FilePath,
+                    frame.LineNumber));
+
+            if (exception is object && _logger is object)
+            {
+                _logger.FailedToReadStackTraceInfo(exception);
```

Comment:
Should this include the original exception as well? Is it OK to include the `Exception` object in the structured log?